tests/components/hardkernel/test_hardware.py

```diff
@@ -0,0 +1,57 @@
+"""Test the Hardkernel hardware platform."""
+import pytest
+
+from homeassistant.components.hardkernel.const import DOMAIN
+from homeassistant.components.hassio import DATA_OS_INFO
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from tests.common import MockModule, mock_integration
+
+
+async def test_hardware_info(hass: HomeAssistant, hass_ws_client) -> None:
+    """Test we can get the board info."""
+    mock_integration(hass, MockModule("hassio"))
+    hass.data[DATA_OS_INFO] = {"board": "odroid-n2"}
+
+    assert await async_setup_component(hass, DOMAIN, {})
```

Comment:
Doesn't the integration require a config entry to be set up properly?

Reply:
Yes, it should have a config entry.
I think what happens here is that the hardware platform is loaded when we try to setup the integration through `async_setup_component`, so the test passes even though the integration is not setup correctly.

The test was copied from the raspberry_pi integration which was changed from not requiring a config entry to requiring a config entry,
I'll fix this test here + the raspberry_py test in a separate PR.